apis/tempo/v1alpha1/microservices_types.go

```diff
@@ -76,6 +76,63 @@ type MicroservicesSpec struct {
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Replication Factor"
 	ReplicationFactor int `json:"replicationFactor,omitempty"`
+
+	// Tenants defines the per-tenant authentication and authorization spec.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Tenants Configuration"
+	Tenants *TenantsSpec `json:"tenants,omitempty"`
+}
+
+// MicroservicesStatus defines the observed state of Microservices.
+type MicroservicesStatus struct {
+	// Version of the managed Tempo instance.
+	// +optional
+	TempoVersion string `json:"tempoVersion,omitempty"`
+
+	// Conditions of the Tempo deployment health.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors="urn:alm:descriptor:io.kubernetes.conditions"
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// ModeType is the authentication/authorization mode in which Tempo Gateway
+// will be configured.
+//
+// +kubebuilder:validation:Enum=dynamic
+type ModeType string
+
+const (
+	// Dynamic mode delegates the authorization to a third-party OPA-compatible endpoint.
+	Dynamic ModeType = "dynamic"
```

Comment:
Are there any other `ModeType` values? Do we think to support them?

Reply:
yes, there are more coming.